apps/src/templates/instructions/InlineAudio.jsx

```diff
@@ -169,6 +175,19 @@ class InlineAudio extends React.Component {
   playAudio() {
     this.getAudioElement().play();
     this.setState({ playing: true });
+    firehoseClient.putRecord(
+      {
+        study: 'tts-play',
+        study_group: 'v1',
+        event: 'play',
+        data_string: this.props.src,
```

Comment:
Is it straightforward to map audio file src back to the message being played?

Reply:
If I understand your question correctly:
The src will be something like: https://tts.code.org/sharon22k/180/100/dcd9c0d72e54896a7d9a3399997885f1/courseC_maze_programming8_predict1_2018.mp3

Each level has a corresponding TTS file, so we can associate the above audio url with the level: courseC_maze_programming8_predict1_2018